megaavr/libraries/Logic/src/Logic.cpp

```diff
@@ -64,13 +64,13 @@ void CustomLogic::init(block_t &block)
   {
     if(block.output_swap == out::pin_swap)
     {
-      PORTMUX.CCLROUTEA |= (block.output_swap << block.block_number);
-      block.PORT->DIR |= PIN6_bm;
+      PORTMUX.CCLROUTEA |= (1 << block.block_number);
```

Comment:
So no need to |= the PINSET register? Will this affect the other pins on this port?

Reply:
The family datasheet of meagAVR-0 series says

> 15.3.2.1 Basic Functions
>
> ....
>
> To use pin number n as an output only, write bit n of the PORTx.DIR register to '1'. This can be done by writing bit n in the PORTx.DIRSET register to '1', which will avoid disturbing the configuration of other pins in that group. The nth bit in the PORTx.OUT register must be written to the desired output value.
>
> Similarly, writing a PORTx.OUTSET bit to '1' will set the corresponding bit in the PORTx.OUT register to '1'. Writing a bit in PORTx.OUTCLR to '1' will clear that bit in PORTx.OUT to zero. Writing a bit in PORTx.OUTTGL or PORTx.IN to '1' will toggle that bit in PORTx.OUT.
>
> To use pin n as an input, bit n in the PORTx.DIR register must be written to '0' to disable the output driver. This can be done by writing bit n in the PORTx.DIRCLR register to '1', which will avoid disturbing the configuration of other pins in that group. The input value can be read from bit n in register PORTx.IN as long as the ISC bit is not set to INPUT_DISABLE.
>
> Writing a bit to '1' in PORTx.DIRTGL will toggle that bit in PORTx.DIR and toggle the direction of the corresponding pin.